### ISSUE TITLE
What is the correct loss reduction to be used when training on multiple GPU?

### ISSUE THREAD

**Issue by ydennisy** [tensorflow/recommenders#222]
Hi!

What should the the loss reduction setting when training across multiple GPUs, this is specific to the ranking task:

```
        self.rating_task = tfrs.tasks.Ranking(
            loss=tf.keras.losses.MeanSquaredError(                
                   reduction=tf.keras.losses.Reduction.SUM
            ),
            metrics=[
                tf.keras.metrics.RootMeanSquaredError(),
            ],
        )
```

The docs [here](https://www.tensorflow.org/tutorials/distribute/custom_training) provide a few options.

Thanks in advance!

**Comment by ydennisy** [comment]
@maciejkula any chance you could take a quick look at this please :)

**Comment by maciejkula** [comment]
Hey - I don't think any loss reduction is necessarily correct, but using `SUM` and dividing by the global batch size (per replica batch size times the number of replicas) seems the canonical solution.

**Comment by ydennisy** [comment]
@maciejkula thanks for your help again!
